src/libstd/sys_common/mod.rs

```diff
@@ -57,6 +57,7 @@ pub mod mutex;
           target_os = "cloudabi",
           target_os = "hermit",
           target_arch = "wasm32",
+          feature = "restricted-std",
```

Comment:
Would it be possible to remove this `#[cfg]` entirely and simply always include the module? If there's warnings about unused definitions we could add `#![allow(warnings)]`, or we could conditionally do that so one target, like unix, emits warnings but everything else just ignores warnings.

Reply:
The implementation is not compatible on windows. The Ext traits conflict, and FromInner isn't implemented.

What about changing it to `cfg(not(windows))`?

Reply:
Ah ok yeah if that's the case then yeah `cfg(not(windows))` I think is a better `#[cfg]`, since that should be applicable for basically everything else.